common/persistence/sql/workflowStateMaps.go

```diff
@@ -483,25 +455,17 @@ func updateSignalInfos(
 			return serviceerror.NewInternal(fmt.Sprintf("Failed to update signal info. Failed to execute update query. Error: %v", err))
 		}
 	}
-	if deleteInfo == nil {
-		return nil
-	}
-	result, err := tx.DeleteFromSignalInfoMaps(sqlplugin.SignalInfoMapsDeleteFilter{
-		ShardID:     shardID,
-		NamespaceID: namespaceID,
-		WorkflowID:  workflowID,
-		RunID:       runID,
-		InitiatedID: deleteInfo,
-	})
-	if err != nil {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update signal info. Failed to execute delete query. Error: %v", err))
-	}
-	rowsAffected, err := result.RowsAffected()
-	if err != nil {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update signal info. Failed to verify number of rows deleted. Error: %v", err))
-	}
-	if int(rowsAffected) != 1 {
```

Comment:
Can we have an equivalent check for new logic?  Basically each tx should result in one row update.

Reply:
all SQL specific tests are here, including number of rows affected (which I probably will remove in the future):
https://github.com/temporalio/temporal/blob/master/common/persistence/sql/sqlplugin/tests

here, checking the number of rows is not necessary (deletion being idempotent)